--- /home/steve/Downloads/cinnamon-common/Tina/usr/share/cinnamon/js/misc/fileUtils.js	2019-09-04 10:48:15.000000000 +0100
+++ /home/steve/Downloads/cinnamon-common/Tina/usr/share/cinnamon/js/misc/fileUtils.js.patched	2019-11-07 05:01:44.942203207 +0000
@@ -81,33 +81,12 @@
     deleteGFile(dir);
 }
 
+// Patched by fileUtils_0.1.patch
 function getUserDesktopDir() {
-
-    // Didn't find a function returning the user desktop dir, so parsing the user-dirs.dirs file to get it
-    let userdirsFile = Gio.file_new_for_path(GLib.get_home_dir()+"/.config/user-dirs.dirs");
-    let path;
-    if (userdirsFile.query_exists(null)){
-        try{
-            let data = userdirsFile.load_contents(null);
-            let dataDic = new Array();
-            let lines = data[1].toString().split("\n");
-            for (var i in lines){
-                if (!lines[i] || lines[i][0]=="#") continue;
-                let line = lines[i].split("=", 2);
-                if (line.length==2){
-                    dataDic[line[0]] = line[1];
-                }
-            }
-            if (dataDic["XDG_DESKTOP_DIR"])
-                path = dataDic["XDG_DESKTOP_DIR"].substring(1, dataDic["XDG_DESKTOP_DIR"].length-1).replace(/\$HOME/, GLib.get_home_dir());
-            else
-                path = GLib.get_home_dir() + '/Desktop';
-        }catch(e){
-            path = GLib.get_home_dir() + '/Desktop';
-        }
-    }else path = GLib.get_home_dir() + '/Desktop';
-    let file = Gio.file_new_for_path(path);
-    if (file.query_exists(null)) return path;
+    GLib.reload_user_special_dirs_cache();
+    let desktopPath = GLib.get_user_special_dir(GLib.UserDirectory.DIRECTORY_DESKTOP);
+    let desktopFile = Gio.file_new_for_path (desktopPath);
+    if (desktopFile.query_exists(null)) return desktopPath;
     else return null;
 }
 
